tests/test_response.py

```diff
@@ -11,3 +13,31 @@ def test_response_set_content_type_set():
 def test_response_set_content_type_not_set():
     resp = falcon.Response()
     assert 'content-type' not in resp._headers
+
+
+def test_response_get_headers():
```

Comment:
Would you mind adding some tests that verify that appending a camel-cased header results in a lowercased header, since that is part of the contract? We should also test that attempting to look up a value using the same non-lowercased form used to set the header raises an instance of `KeyError`.